src/yunohost/backup.py

```diff
@@ -49,111 +51,199 @@
-                  no_compress=False, ignore_hooks=False, hooks=[],
-                  ignore_apps=False, apps=[]):
+class Archive:
     """
-    Create a backup local archive
-
-    Keyword arguments:
-        name -- Name of the backup archive
-        description -- Short description of the backup
-        output_directory -- Output directory for the backup
-        no_compress -- Do not create an archive file
-        hooks -- List of backup hooks names to execute
-        ignore_hooks -- Do not execute backup hooks
-        apps -- List of application names to backup
-        ignore_apps -- Do not backup apps
-
+    This class prepares files to backup and apply one or several backup method
+    on it.
     """
-    # TODO: Add a 'clean' argument to clean output directory
-    tmp_dir = None
-    env_var = {}
 
-    # Validate what to backup
-    if ignore_hooks and ignore_apps:
-        raise MoulinetteError(errno.EINVAL,
-            m18n.n('backup_action_required'))
-
-    # Validate and define backup name
-    timestamp = int(time.time())
-    if not name:
-        name = time.strftime('%Y%m%d-%H%M%S')
-    if name in backup_list()['archives']:
-        raise MoulinetteError(errno.EINVAL,
-            m18n.n('backup_archive_name_exists'))
-
-    # Validate additional arguments
-    if no_compress and not output_directory:
-        raise MoulinetteError(errno.EINVAL,
-            m18n.n('backup_output_directory_required'))
-    if output_directory:
-        output_directory = os.path.abspath(output_directory)
-
-        # Check for forbidden folders
-        if output_directory.startswith(ARCHIVES_PATH) or \
-           re.match(r'^/(|(bin|boot|dev|etc|lib|root|run|sbin|sys|usr|var)(|/.*))$',
-                    output_directory):
-            raise MoulinetteError(errno.EINVAL,
-                m18n.n('backup_output_directory_forbidden'))
-
-        # Create the output directory
-        if not os.path.isdir(output_directory):
-            logger.debug("creating output directory '%s'", output_directory)
-            os.makedirs(output_directory, 0750)
-        # Check that output directory is empty
-        elif no_compress and os.listdir(output_directory):
-            raise MoulinetteError(errno.EIO,
-                m18n.n('backup_output_directory_not_empty'))
-
-        # Do not compress, so set temporary directory to output one and
-        # disable bind mounting to prevent data loss in case of a rm
-        # See: https://dev.yunohost.org/issues/298
-        if no_compress:
-            logger.debug('bind mounting will be disabled')
-            tmp_dir = output_directory
-            env_var['CAN_BIND'] = 0
-    else:
-        output_directory = ARCHIVES_PATH
-
-    # Create archives directory if it does not exists
-    if not os.path.isdir(ARCHIVES_PATH):
-        os.mkdir(ARCHIVES_PATH, 0750)
+    def __init__(self, name=None, description='', collect_dir=None):
+        self.info = {
```

Comment:
You right, historically it was in this dict but it could be a good idea to change that.